src/planscape/forsys/cluster_stands_test.py

```diff
@@ -0,0 +1,453 @@
+            2: {0: {'foo': 0.5, 'bar': 0.3},  1: {'foo': 0.1, 'bar': 0.3}},
+            3: {0: {'foo': 0.5, 'bar': 0.3},  1: {'foo': 0.1, 'bar': 0.3}},
+        }
+        priority_weights = {
+            'foo': 10,
+            'bar': 5
+        }
+
+        # Baseline: preference for a blob over a line.
+        clustered_stands = ClusteredStands(
+            pixel_dist_to_condition_values, 4, 2, 1, priority_weights, 1, 3)
+        self.assertDictEqual(clustered_stands.clusters_to_stands,
+                             {0: [(0, 0), (0, 1), (1, 0), (1, 1)],
+                              1: [(2, 1), (3, 1)],
+                              2: [(2, 0), (3, 0)]})
+
+        # Scaling pixel_index_weight by a factor of 1/100: preference for a line
+        # over a blob.
+        clustered_stands_piw0p01 = ClusteredStands(
+            pixel_dist_to_condition_values, 4, 2, 1, priority_weights, 1e-2, 3)
+        self.assertDictEqual(clustered_stands_piw0p01.clusters_to_stands,
+                             {0: [(0, 0), (1, 0), (2, 0), (3, 0)],
+                              1: [(2, 1), (3, 1)],
+                              2: [(0, 1), (1, 1)]})
+
+        # Scaling priority weights by a factor of 100: preference for a blob
+        # over a line (no change from baseline).
+        priority_weights_100x = {
+            'foo': 1000,
+            'bar': 500
+        }
+        clustered_stands_pw10 = ClusteredStands(
+            pixel_dist_to_condition_values, 4, 2, 1, priority_weights_100x, 1,
+            3)
+        self.assertDictEqual(clustered_stands_pw10.clusters_to_stands,
+                             {0: [(0, 0), (0, 1), (1, 0), (1, 1)],
+                              1: [(2, 1), (3, 1)],
+                              2: [(2, 0), (3, 0)]})
+
+    def test_normalizes_condition_scores(self) -> None:
+        pixel_dist_to_condition_values = {
+            0: {0: {'foo': 0.5, 'bar': 0.3},  1: {'foo': 0.4, 'bar': 0.3}},
+            1: {0: {'foo': 0.5, 'bar': 0.3},  1: {'foo': 0.4, 'bar': 0.3}},
+            2: {0: {'foo': 0.5, 'bar': 0.3},  1: {'foo': 0.1, 'bar': 0.3}},
+            3: {0: {'foo': 0.5, 'bar': 0.3},  1: {'foo': 0.1, 'bar': 0.3}},
+        }
+        priority_weights = {
+            'foo': 10,
+            'bar': 5
+        }
+
+        # Baseline: preference for a blob over a line.
+        clustered_stands = ClusteredStands(
+            pixel_dist_to_condition_values, 4, 2, 1, priority_weights, 1, 3)
+        self.assertDictEqual(clustered_stands.clusters_to_stands,
+                             {0: [(0, 0), (0, 1), (1, 0), (1, 1)],
+                              1: [(2, 1), (3, 1)],
+                              2: [(2, 0), (3, 0)]})
+
+        # Scaling pixel_index_weight by a factor of 1/100: preference for a line
+        # over a blob.
+        clustered_stands_piw0p01 = ClusteredStands(
+            pixel_dist_to_condition_values, 4, 2, 1, priority_weights, 1e-2, 3)
```

Comment:
for normalization, how do we determine when to set a scaling value?

Reply:
Do you mean how we should determine pixel_index_weight?

That's tbd - I don't have recommendations yet but have been thinking about it. It seems to be a function of how large we want clusters to be, which would be a function of the number of pixels and the desired number of clusters.

We can estimate forsys performance benchmarks for a specific number of clusters, and then perhaps choose pixel_index_weight as a function of the number of pixels.

Reply:
"how large we want clusters to be" - should clusters be proportional to the size of the planning area? or should we have a consistent cluster size to ensure forsys clusters are grouped consistently across different runs?